controllers/runnerreplicaset_controller.go

```diff
@@ -133,52 +142,11 @@ func (r *RunnerReplicaSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		registrationOnlyRunnerExists = true
 	}
 
-	// On scale to zero, we must have fully registered registration-only runner before we start deleting other runners, hence `desired == 0`
-	// On scale from zero, we must retain the registratoin-only runner until one or more other runners get registered, hence `registrationOnlyRunnerExists && available == 0`.
-	// On RunnerReplicaSet creation, it have always 0 replics and no registration-only runner.
-	// In this case We don't need to bother creating a registration-only runner which gets deleted soon after we have 1 or more available repolicas,
-	// hence it's not `available == 0`, but `registrationOnlyRunnerExists && available == 0`.
-	// See https://github.com/actions-runner-controller/actions-runner-controller/issues/516
-	registrationOnlyRunnerNeeded := desired == 0 || (registrationOnlyRunnerExists && current == 0)
+	if registrationOnlyRunnerExists {
+		if err := r.Client.Delete(ctx, &registrationOnlyRunner); err != nil {
```

Comment:
I believe we'd better maintain this reg-only-runner clean up logic for a while(perhaps until 0.22 or 0.23) so that we can prevent remaining and dangling reg-only runners in everyone's cluster.

Reply:
Makes sense, probably worth having a comment, e.g.
```
// TODO: remove registration runner check later (v0.23.0 or v0.24.0)
```

Reply:
Thanks! Added 807e630 for the additional comment.